src/main/java/com/github/steveice10/mc/protocol/data/message/EntityHoverMessage.java

```diff
@@ -0,0 +1,70 @@
+package com.github.steveice10.mc.protocol.data.message;
+
+import com.github.steveice10.mc.protocol.data.message.style.MessageStyle;
+import lombok.EqualsAndHashCode;
+import lombok.NonNull;
+
+import java.util.List;
+
+@EqualsAndHashCode(callSuper = true)
+public class EntityHoverMessage extends Message {
+    public static class Builder extends Message.Builder<Builder, EntityHoverMessage> {
+        @NonNull
+        private String type = "";
+        @NonNull
+        private String id = "";
+        private Message name;
+
+        public Builder type(@NonNull String type) {
+            this.type = type;
+            return this;
+        }
+
+        public Builder id(@NonNull String id) {
+            this.id = id;
+            return this;
+        }
+
+        public Builder name(@NonNull Message name) {
```

Comment:
If this is optional like the field indicates, parameter should be nullable as well.

Reply:
Fixed